### ISSUE TITLE
Needs to be started as network user/group in openwrt

### ISSUE THREAD

**Issue by timdodge** [awilliams/wifi-presence#3]
Hi, thanks for the integration - it was included in openwrt head today.

In order for it to work I needed to alter the init script to start it as the network user - it can't access the control interface otherwise.

**Comment by McGiverGim** [comment]
I've the same problem. Could you explain how can I do it?

**Comment by timdodge** [comment]
Edit /etc/init.d/wifi-presence
Add to start_service():
>procd_set_param user network
>procd_set_param group network

**Comment by McGiverGim** [comment]
I added the two parameters. I don't have the error anymore, but I think I have something bad configured. I can't see anything at the MQTT server. I will look again to the configuration.
Thanks for the help!
@awilliams could you add this two parameters to the OpenWrt package?

**Comment by awilliams** [comment]
I've only seen processes run as `root` on OpenWrt unless explicitly set otherwise. For this reason, I've had no permissions issues. Is `wifi-presence` being run as `root` for you and you still see the permissions issues with the control sockets?

Instead of hardcoding `network`, I'd like to make it configurable (similar to [kcptun](https://github.com/openwrt/packages/blob/51a5ec60690674d4525dd14bade280110743e61b/net/kcptun/files/kcptun.init#L74) for example). Would that solve the issue you're experiencing?

**Comment by McGiverGim** [comment]
I can't say too much, I don't have too much experience with Linux or Openwrt. I only installed the package with opkg and I experienced the issue until I added the lines commented by @timdodge.

**Comment by timdodge** [comment]
I'm not an expert on socket "files", but the permission on the control sockets are:
>root@wap:~# ls -la /var/run/hostapd/wlan0
>srwxrwx---    1 network  network          0 Apr 11 12:25 /var/run/hostapd/wlan0

When running as the root user, wifi-presence can't access them. I've tried catting the socket as root and just get a "No such device or address" error back. I guess that as a minimum, the wifi-presence user needs to be in the network group.

**Comment by jmccrohan** [comment]
@awilliams I encountered this issue also, and was resolved using the changes suggested by @timdodge. Feel free to ping me if you want to make these changes directly in OpenWrt.

**Comment by awilliams** [comment]
I've created a PR that I think will address this issue: [PR #6](https://github.com/awilliams/wifi-presence/pull/6).

Please try it out if you're able to. After updating the `/etc/init.d/wifi-presence` script, add the following to `/etc/config/wifi-presence` to run the process as network / network:

```
         option runAsUser 'network'
         option runAsGroup 'network'
```

**Comment by jmccrohan** [comment]
@awilliams can you share a bit more detail on your OpenWrt builds? Are you running an old build/snapshot, or a target where `seccomp` is unavailable? Given that `seccomp` is enabled by default on most (all?) targets, I think that the user/group options should be uncommented in the config file.
https://github.com/awilliams/wifi-presence/pull/6/commits/ad2b6bde4e7dc99c449737d288e488a4dd2ff684#diff-5000e54b7b529e6c12b270a4c9335886f97c9b4065dda3d789740f4e7a042aa5R52-R53

**Comment by awilliams** [comment]
@jmccrohan Thanks for letting me know that `seccomp` is enabled by default. I'm using a custom build of OpenWrt for my routers, so wasn't aware. 

What do you think about the following idea? Modify the changes in #6 to auto-detect the proper user if `runAsUser`/`runAsGroup` aren't explicitly defined. In other words, instead of defaulting to `root`, do something like stat the `/var/run/hostapd/wlan0` sockets and run wifi-presence as the owner of that file.

**Comment by jmccrohan** [comment]
@awilliams Up to you how you implement it. A simple comment might in the config file re `root`/`network` users might suffice and is easier to maintain. Key is to get this fixed and merged into OpenWrt so that new users don't get bitten by this issue. :-)

**Comment by awilliams** [comment]
I think the recent change in #6 will work fine. 

It allows setting a user/group in `/etc/config/wifi-presence`. If no specific values are set, then the init script will look at the owner of the sockets in `/var/run/hostapd` and use that user as the user/group to start wifi-presence. 

If you're able @jmccrohan, please review. Once merged in this repo, I'll make the same change to `openwrt/packages`.

**Comment by jmccrohan** [comment]
Thanks, LGTM.

Open a PR against `openwrt-packages` and ping me to merge.

**Comment by awilliams** [comment]
https://github.com/openwrt/packages/pull/18573

**Comment by awilliams** [comment]
Fixed by openwrt/packages#18573
